src/methods/function.js

```diff
@@ -269,17 +269,30 @@ function InternalConstruct(
     function map(value: Value) {
       if (value === realm.intrinsics.__bottomValue) return value;
 
-      if (value instanceof AbstractValue && value.kind === "conditional") {
-        const [condition, consequent, alternate] = value.args;
-        return realm.evaluateWithAbstractConditional(
-          condition,
-          () => realm.evaluateForEffects(() => map(consequent), undefined, "AbstractValue/conditional/true"),
-          () => realm.evaluateForEffects(() => map(alternate), undefined, "AbstractValue/conditional/false")
-        );
+      if (value instanceof AbstractValue && !(value instanceof AbstractObjectValue)) {
+        if (value.kind === "conditional") {
```

Comment:
It is not clear to me why this case now need value to not be an abstract object value.